### PR TITLE
 [ADD] pos_warning_exiting : new module to add extra checks before exiting from Point of sale Front office UI

### PR DESCRIPTION
**Pull request by legalsylvain**
This module was written to extend the functionality of Odoo Point Of Sale, adding extra checks when exiting
from the Point of Sale front office UI.

* If the cashier doesn't have a connection to the Odoo server, a message will
  be displayed.

image

* If the current session contains unpaid draft orders, a message will be displayed.

image

ping @quentinDupont, @grap 
#GRAPOCA deck/#/board/144/card/1215

### CODE REVIEW

File: pos_warning_exiting/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    "license": "AGPL-3",
+    "depends": [
+        "point_of_sale",
+    ],
+    "data": [
+        "views/templates.xml",
+    ],
+    "images": [
+        # "static/description/error_message.png",
+    ],
```
Comment (ivantodorovich) [inline pos_warning_exiting/__manifest__.py]:
```suggestion
```

To remove

File: pos_warning_exiting/static/src/js/gui.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        var draft_orders_description = [];
+        var order_jsons = this.pos.db.get_unpaid_orders();
+
+        order_jsons.forEach(function (order_json) {
+            if (order_json.pos_session_id === self.pos.pos_session.id && order_json.lines.length) {
+                var order_description = _t("Order #") + order_json.sequence_number;
+                // Lazy depedency to pos_restaurant
+                // to display table information
+                if (order_json.table) {
+                    order_description += _t(" - Table : ") + order_json.table.name;
```
Comment (ivantodorovich) [inline pos_warning_exiting/static/src/js/gui.js]:
```suggestion
                    order_description += _t(" - Table: ") + order_json.table.name;
```

File: pos_warning_exiting/static/src/js/gui.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                }
+                draft_orders_description.push(order_description);
+            }
+        });
+
+        if (draft_orders_description.length) {
+            this.show_popup("confirm", {
+                "title": _t("Draft Orders"),
+                "body":  _t(
+                    "You have some draft unpaid orders." +
```
Comment (ivantodorovich) [inline pos_warning_exiting/static/src/js/gui.js]:
```suggestion
                     "You have some draft unpaid orders." +
```

File: pos_warning_exiting/static/src/js/gui.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        //     if (order.orderlines.length) {
+        //         var order_description = _t("Order #") + order.sequence_number;
+        //         // Lazy depedency to pos_restaurant
+        //         // to display table information
+        //         if (order.table) {
+        //             order_description += _t(" - Table : ") + order.table.name;
+        //         }
+        //         draft_orders_description.push(order_description);
+        //     }
+        // });
```
Comment (ivantodorovich) [inline pos_warning_exiting/static/src/js/gui.js]:
```suggestion
```
Remove if not used

**Review by ivantodorovich — APPROVED** [review]
Minor comments, lgtm :+1: !!

File: pos_warning_exiting/__manifest__.py (hunk 1 above)
Comment (legalsylvain) [reply pos_warning_exiting/__manifest__.py]: better to add the reference to the 2 images.

**Comment by legalsylvain** [comment]
> Minor comments, lgtm +1 !!

Thanks for your review ! I took into account your suggestions.

File: pos_warning_exiting/static/src/js/gui.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    var utils = require('web.utils');
+    var gui = require("point_of_sale.gui");
+
+    var _t = core._t;
+    var round_pr = utils.round_precision;
+
+
+    var _super_close_ = gui.Gui.prototype.close;
+
+    gui.Gui.prototype.close = function(){
```
Comment (ivantodorovich) [inline pos_warning_exiting/static/src/js/gui.js:21]: Is there a reason for using prototype instead of `gui.Gui.include`? :thinking:

Comment (legalsylvain) [reply pos_warning_exiting/static/src/js/gui.js:21]: What is the other way ? 
I mean : 

local ``close`` calls  local ``check_connection_and_close`` that calls super of ``close``. 
Using prototype seems necessary.

Comment (ivantodorovich) [reply pos_warning_exiting/static/src/js/gui.js:21]: Oh yes, I see that now 👍🏻

**Pull request merged**